components/odh-notebook-controller/go.mod

```diff
@@ -1,18 +1,22 @@
 module github.com/opendatahub-io/kubeflow/components/odh-notebook-controller
 
-go 1.21
+go 1.22
 
-toolchain go1.21.9
+toolchain go1.23.2
```

Comment:
Is this go and other dependency update related to this oTel addition? If not, I slightly tend to have it in a separate PR/commit? :thinking:

Reply:
oversight, sorry about that, will try to push it back to 2.21

Reply:
```
go mod tidy -go=1.21.9
go: go.opentelemetry.io/auto/sdk@v1.1.0 requires go@1.22.0, but 1.21.9 is requested
```

Reply:
I see, in that case, I would probably introduce the golang bump update first after some agreement and only then introduced these tests?

Reply:
probably, I tried making it settle on 1.21 and I could not make it work without going way back with otel

so yes, let's switch to 1.22 first